algorithms/benches/merkle_tree/merkle_tree.rs

```diff
@@ -32,13 +32,16 @@ use criterion::Criterion;
 const SETUP_MESSAGE: &str = "merkle_tree_bhp_benchmark";
 
 const NUM_WINDOWS: usize = 3;
-const WINDOW_SIZE: usize = 57;
+const LEAF_WINDOW_SIZE: usize = 29;
+const TWO_TO_ONE_WINDOW_SIZE: usize = 57;
 const TREE_DEPTH: usize = 32;
 
-type H = BHPCRH<EdwardsProjective, NUM_WINDOWS, WINDOW_SIZE>;
-type P = MerkleTreeParameters<H, TREE_DEPTH>;
+type LeafCRH = BHPCRH<EdwardsProjective, NUM_WINDOWS, LEAF_WINDOW_SIZE>;
+type TwoToOneCRH = BHPCRH<EdwardsProjective, NUM_WINDOWS, TWO_TO_ONE_WINDOW_SIZE>;
+type P = MerkleTreeParameters<LeafCRH, TwoToOneCRH, TREE_DEPTH>;
 
 const NUM_ENTRIES: &[usize] = &[10, 100, 1000, 10000];
+const REBUILD_SIZES: &[usize] = &[10, 100, 1000];
```

Comment:
I'd also suggest including some cases ending up with a tree with `2^n` entries here to also measure the worst-case scenario